src/path.rs

```diff
@@ -0,0 +1,140 @@
+use Renderable;
+use value::Value;
+use context::Context;
+use token::Token;
+use token::Token::*;
+use error::{Error, Result};
+
+#[derive(Debug)]
+pub struct IdentifierPath {
+    value: String,
+    indexes: Vec<Value>,
+}
+
+impl Renderable for IdentifierPath {
+    fn render(&self, context: &mut Context) -> Result<Option<String>> {
+        println!("{:?}", self);
+        let value = context
+            .get_val(&self.value)
+            .ok_or_else(|| Error::Render(format!("{} not found in context", self.value)))?
+            .clone();
+
+        let mut counter = self.indexes.len();
+        let result = self.indexes.iter().fold(Ok(&value), |value, index| {
+            // go through error
+            let value = if let Ok(value) = value {
+                value
+            } else {
+                return value;
+            };
+            counter -= 1;
+            match (value, index) {
+                (&Value::Array(ref value), &Value::Num(ref x)) => {
+                    // at the first condition only is_normal is not enough
+                    // because zero is not counted normal
+                    if (*x != 0f32 && !x.is_normal()) || *x < 0f32 ||
+                       x.round() > (::std::usize::MAX as f32) {
```

Comment:
Does liquid support negative indexing like Python?  Does it support slicing?

(and this code is one of the reasons why I'm considering native integer support)

Reply:
Neither of docs I see say this explicitly, but a simple test in jekyll shows that negative indesinx is allowed, ranges(I've tried `[0..1]` `[0:1]` `[0-1]`) are not

Reply:
Thanks for looking into this!